rust/psibase_macros/psibase-macros-lib/src/tests/basics.rs

```diff
@@ -0,0 +1,51 @@
+#[cfg(test)]
+mod tests {
+    use crate::service_macro::service_macro_impl;
+
+    #[test]
+    #[should_panic]
+    // Events can only have one of 3 values: history, ui, or merkle
+    // How to test the error text?
+    fn test_event_attribute_parsing_event_type() {
+        use quote::quote;
+
+        let attr = quote! {};
+        let before = quote! {
+            mod service {
+                #[action]
+                fn add(a: i32, b: i32) -> i32 {
+                    a + b
+                }
+                #[event(historyui)]
+                fn something() {
+
+                }
+            }
+        };
+
+        let after = service_macro_impl(attr, before);
```

Comment:
Doesn't this cause a warning?

Reply:
```
warning: unused variable: `after`
  --> psibase_macros/psibase-macros-lib/src/tests/basics.rs:26:13
   |
26 |         let after = service_macro_impl(attr, before);
   |             ^^^^^ help: if this is intentional, prefix it with an underscore: `_after`
   |
   = note: `#[warn(unused_variables)]` on by default
```

Reply:
I get this by running `cargo test`

Reply:
oh, sorry. relied on my linter, which did *not* give that warning (oh, because it cfg(test). but yes, the var, yes. There's a warning. I added an underscore to stop it.